sky/resources.py

```diff
@@ -741,11 +753,12 @@ def get_valid_regions_for_launchable(self) -> List[clouds.Region]:
         ssh_proxy_command dict with region names as keys).
         """
         assert self.is_launchable(), self
-
-        regions = self._cloud.regions_with_offering(self._instance_type,
-                                                    self.accelerators,
-                                                    self._use_spot,
-                                                    self._region, self._zone)
+        assert self.cloud is not None, 'Cloud must be specified'
+        assert self.instance_type is not None, 'Instance type must be specified'
+        regions = self.cloud.regions_with_offering(self.instance_type,
+                                                   self.accelerators,
+                                                   self.use_spot, self.region,
+                                                   self.zone)
```

Comment:
cloud, instance_type, use_spot, region, and zone are straight lookup properties, with no additional logic, so this is functionally the same.
`self.accelerators` was already used here instead of `self._accelerators`, so also no change there.

Reply:
To be clear, I was just marking down that I checked that this change is okay. Either way is ok.